GG/src/Wnd.cpp

```diff
@@ -506,8 +536,11 @@ void Wnd::SetMinSize(const Pt& sz)
     if (Width() < m_min_size.x || Height() < m_min_size.y)
         Resize(Pt(std::max(Width(), m_min_size.x), std::max(Height(), m_min_size.y)));
     // The previous Resize() will call ChildSizeOrMinSizeChanged() itself if needed
-    else if (m_containing_layout && min_size_changed && !dynamic_cast<Layout*>(this))
-        m_containing_layout->ChildSizeOrMinSizeChanged();
+    else {
+        auto containing_layout = LockAndResetIfExpired(m_containing_layout);
+        if (containing_layout && min_size_changed && !dynamic_cast<Layout*>(this))
```

Comment:
you want to call reset if expired regardless of the value of `min_size_changed` ?